app/models/ability.rb

```diff
@@ -0,0 +1,42 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+
+    user ||= User.new
+
+    can :manage, Post do |post|
+      post.user == user && user.persisted?
+    end
+
+    can :destroy, Comment do |comment|
```

Comment:
Since you also have comments#create implemented, consider: 
    alias_action :create, :read, :udpate, :destroy, :to => :crud
  can :crud, Comment do |comment|
      # allow the post owner and the comment owner to crud
      (comment.post.user == user || comment.user == user) && user.persisted?
    end